Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionControllerConfigurationCocoa.mm

```diff
@@ -86,6 +86,20 @@
     return m_webViewConfiguration.get();
 }
 
+String WebExtensionControllerConfiguration::declarativeNetRequestStoreDirectory()
+{
+    if (!m_declarativeNetRequestStoreDirectory.isEmpty())
+        return m_declarativeNetRequestStoreDirectory;
+
+    if (!storageIsPersistent()) {
+        m_declarativeNetRequestStoreDirectory = FileSystem::createTemporaryDirectory(@"DeclarativeNetRequest");
+        return m_declarativeNetRequestStoreDirectory;
+    }
+
+    m_declarativeNetRequestStoreDirectory = FileSystem::pathByAppendingComponent(m_storageDirectory, "DeclarativeNetRequest"_s);
+    return m_declarativeNetRequestStoreDirectory;
```

Comment:
Should move `FileSystem::makeAllDirectories` here before the return since it is one time and cached.